### PR TITLE
backport: add event subscription options and defaults config (#7930)

### PR DESCRIPTION
**Pull request by mmsqe**
For the full description, see #7930

---

#### PR checklist

- [ ] Tests written/updated, or no tests needed
- [ ] `CHANGELOG_PENDING.md` updated, or no changelog entry needed
- [ ] Updated relevant documentation (`docs/`) and code comments, or no
      documentation updates needed

### CODE REVIEW

**Comment by mmsqe** [comment]
@sergio-mena may I ask if I could merge the rest backports into one for faster review?

**Comment by sergio-mena** [comment]
> @sergio-mena may I ask if I could merge the rest backports into one for faster review?

Yes, makes sense. We are at the end of the quarter and I can't guarantee a timely review (I'm reviewing this PR now).

However, for the sake of clarity and traceability, please include each backported PR into its own commit in the new PR.
We will not squash your PR once approved, but will rebase it to `feature/adr075-backport`.
This way we'll get the best of both worlds (reduced back and forth interactions, but keeping traceability of the different changes).

File: CHANGELOG_PENDING.md [hunk 1]
```diff
@@ -36,6 +36,7 @@ Friendly reminder, we have a [bug bounty program](https://hackerone.com/tendermi
 ### BREAKING CHANGES
 
 - CLI/RPC/Config
+  - [config] \#7930 Add new event subscription options and defaults. (@creachadair)
```
Comment (sergio-mena) [inline CHANGELOG_PENDING.md:39]: I have edited this PR's description to refer to #7930.
So we think it's better that this changelog line refers to #9491.

File: node/node.go [hunk 2]
```diff
@@ -1116,25 +1117,27 @@ func (n *Node) startRPC() ([]net.Listener, error) {
...
-			rpcserver.OnDisconnect(func(remoteAddr string) {
-				err := n.eventBus.UnsubscribeAll(context.Background(), remoteAddr)
-				if err != nil && err != tmpubsub.ErrSubscriptionNotFound {
-					wmLogger.Error("Failed to unsubscribe addr from events", "addr", remoteAddr, "err", err)
-				}
-			}),
-			rpcserver.ReadLimit(config.MaxBodyBytes),
-			rpcserver.WriteChanCapacity(n.config.RPC.WebSocketWriteBufferSize),
-		)
-		wm.SetLogger(wmLogger)
```
Comment (sergio-mena) [inline node/node.go:1135]: Similarly, to my comment in line 1133, wouldn't we want to keep this line?

File: node/node.go [hunk 3]
```diff
@@ -1116,25 +1117,27 @@ func (n *Node) startRPC() ([]net.Listener, error) {
...
-		wmLogger := rpcLogger.With("protocol", "websocket")
-		wm := rpcserver.NewWebsocketManager(rpccore.Routes,
-			rpcserver.OnDisconnect(func(remoteAddr string) {
-				err := n.eventBus.UnsubscribeAll(context.Background(), remoteAddr)
-				if err != nil && err != tmpubsub.ErrSubscriptionNotFound {
-					wmLogger.Error("Failed to unsubscribe addr from events", "addr", remoteAddr, "err", err)
-				}
-			}),
-			rpcserver.ReadLimit(config.MaxBodyBytes),
-			rpcserver.WriteChanCapacity(n.config.RPC.WebSocketWriteBufferSize),
```
Comment (sergio-mena) [inline node/node.go:1133]: This line wasn't on the original PR, so you might want to keep it after this PR.
From a purely "backporting" point of view, it does make a lot of sense to keep this line.

**Review by sergio-mena — COMMENTED** [review]
Thanks for this work

File: node/node.go (hunk 3 above)
Comment (mmsqe) [reply node/node.go:1133]: Thanks for catching, added back.

**Review by sergio-mena — APPROVED** [review]
From a backporting point of view, this is ready for merging

**Pull request merged**